costs/es_request_constructor.go

```diff
@@ -0,0 +1,254 @@
+//   Copyright 2017 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package costs
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"gopkg.in/olivere/elastic.v5"
+)
+
+// paramNameToFuncPtr : map of string keys and functions pointer as values. For each possible param
+// after parsing (removing the ':<TAG_KEY>' in the case of the tag), there is a function associated to it
+// that create the Aggregations needed by this param.
+// If a new param, that is only creating aggregations, needs to be added,
+// a functions with the same prototype as the one below should be added to this map.
+var paramNameToFuncPtr = map[string]func([]string) []paramAggrAndName{
```

Comment:
`func([]string) []paramAggrAndName` is not a descriptive type name, why not alias it ? It makes complex types more readable, for example `map[string]aggregationBuilder`.